react-effects/my-app/src/List.tsx

```diff
@@ -0,0 +1,46 @@
+/* eslint-disable @typescript-eslint/no-unused-vars -- Remove me */
+import { useEffect, useState } from 'react';
+import { readItems, type Item } from './read';
+
+export function List() {
+  const [isLoading, setIsLoading] = useState(true);
+  const [items, setItems] = useState<Item[]>([]);
+  const [error, setError] = useState<unknown>();
+
+  useEffect(() => {
+    async function effecting() {
+      try {
+        const settlingLisMapThrough = await readItems();
+
+        setItems(settlingLisMapThrough);
+      } catch (err) {
+        console.log('Error message:', err);
+        console.log('Cannot get:', setIsLoading(!isLoading));
+      }
+    }
+
+    effecting();
+  }, [items]);
```

Comment:
Remove `items` from the dependency array. It's not a dependency and it will cause unnecessary re-renders.